.pre-commit-config.yaml

```diff
@@ -0,0 +1,31 @@
+repos:
+    -   repo: https://github.com/pycqa/isort
+        rev: 5.8.0
+        hooks:
+        -    id: isort
+             additional_dependencies: ["colorama>=0.4.3"]
+
+    -   repo: https://github.com/psf/black
+        rev: 21.4b0
+        hooks:
+        -   id: black
+
+    -   repo: https://gitlab.com/pycqa/flake8.git
+        rev: 3.8.0
+        hooks:
+        -   id: flake8
+
+    -   repo: https://github.com/pycqa/pydocstyle
+        rev: 6.1.0
+        hooks:
+        -   id: pydocstyle
+            exclude: (tests|docs)
+            additional_dependencies: ["toml"]
+
+    -   repo: https://github.com/pre-commit/mirrors-mypy
+        rev: v0.910
+        hooks:
+        -   id: mypy
+            args: [--strict, --ignore-missing-imports]
+            additional_dependencies: [torch>=1.7, pytorch-lightning>=1.3, pytest>=6, omegaconf>=2.1]
```

Comment:
How did you determine this list of dependencies? A lot of packages have partial type hints, so they should still be installed. The annoying thing about this is that if a dependency gains type hints we won't know. Torchmetrics is definitely missing from this, not sure what else is missing. Personally I would just add all dependencies.

Reply:
You're going to love (hate) this :)

I added libraries that I knew that used type hints and confirmed by running the default mypy. I stopped adding dependencies once I saw the mypy hook and default mypy gave the same result.

Reply:
I'm guessing the reason that not having torchmetrics in that list didn't raise any red flags is that mypy will ignore missing type hints by default. We might just want to add all dependencies (and have a sixth file we need to keep track of all deps in...)